### PR TITLE
Add pcap arm pvi entry

### LINKED ISSUE #20: Add PCAP.ARM pvi entry
Arm needs adding to pvi

https://github.com/PandABlocks/PandABlocks-ioc/blob/9c6495ea39f93a1874524fd94020c53a56014909/src/pandablocks_ioc/ioc.py#L1806-L1816

I suggest adding it as a `SignalRW` with read widget `LED` and write widget `ComboBox` (for now, we will have `ActionButtons` in the future). This should also produce a pvi PV for ophyd.

### PR DESCRIPTION
**Pull request by evalott100**
Closes #20 

Branches off of https://github.com/evalott100/PandABlocks-ioc/tree/make_pv_names_consistent_with_web_gui_names

### CODE REVIEW

File: src/pandablocks_ioc/_pvi.py [hunk 1]
```diff
@@ -61,9 +63,16 @@ def add_pvi_info(
     useComboBox: bool = record_creation_func == builder.mbbOut
 
     if record_creation_func == builder.Action:
-        # TODO: What value do I write? PandA uses an empty string
-        component = SignalX(record_name, record_name, value="")
-        access = "x"
+        if record_name == "PCAP:ARM":
```
Comment (evalott100) [inline src/pandablocks_ioc/_pvi.py:66]: I investigated checking the record for `ONAM: "Arm"` `ZNAM: "Disarm"` but with a `RecordWrapper` this code would be harder to read and have worse typing than simply checking if `record_name == "PCAP:ARM"'

File: src/pandablocks_ioc/_pvi.py [hunk 2]
```diff
@@ -61,9 +63,16 @@ def add_pvi_info(
...
+            component = SignalRW(
+                record_name, record_name, widget=ButtonPanel(), read_widget=LED()
```
Comment (coretl) [inline src/pandablocks_ioc/_pvi.py:68]:
```suggestion
                record_name, record_name, widget=ButtonPanel(actions=dict(Arm=1, Disarm=0)), read_widget=LED()
```

**Review by coretl — COMMENTED** [review]
Just one tweak

**Comment by dmgav** [comment]
I tried running the code from this branch with PandA and I am getting the following error:
```
$ PandABlocks-ioc softioc xf03idc-panda1.nsls2.bnl.local PANDA ./screens
ERROR:Exception while initializing softioc
Traceback (most recent call last):
  File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 132, in create_softioc
    ).result()
  File "/nsls2/users/dgavrilov/conda_envs/2023-2.1-py310-tiled-panda/lib/python3.10/concurrent/futures/_base.py", line 458, in result
    return self.__get_result()
  File "/nsls2/users/dgavrilov/conda_envs/2023-2.1-py310-tiled-panda/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 98, in _create_softioc
    (all_records, all_values_dict, panda_dict) = await create_records(
  File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 1754, in create_records
    (panda_dict, all_values_dict) = await introspect_panda(client)
  File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 179, in introspect_panda
    values, all_values_dict = _create_dicts_from_changes(changes, block_dict)
  File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 266, in _create_dicts_from_changes
    _store_values(block_and_field_name, value, values)
  File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 228, in _store_values
    number_of_blocks = block_info[block_name_number[:-1]].number
KeyError: 'TTLOUT1'
```

**Comment by evalott100** [comment]
@dmgav 

> I tried testing the code from this branch with PandA and I am getting the following error:
> 
> ```
> $ PandABlocks-ioc softioc xf03idc-panda1.nsls2.bnl.local PANDA ./screens
> ERROR:Exception while initializing softioc
> Traceback (most recent call last):
>   File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 132, in create_softioc
>     ).result()
>   File "/nsls2/users/dgavrilov/conda_envs/2023-2.1-py310-tiled-panda/lib/python3.10/concurrent/futures/_base.py", line 458, in result
>     return self.__get_result()
>   File "/nsls2/users/dgavrilov/conda_envs/2023-2.1-py310-tiled-panda/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
>     raise self._exception
>   File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 98, in _create_softioc
>     (all_records, all_values_dict, panda_dict) = await create_records(
>   File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 1754, in create_records
>     (panda_dict, all_values_dict) = await introspect_panda(client)
>   File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 179, in introspect_panda
>     values, all_values_dict = _create_dicts_from_changes(changes, block_dict)
>   File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 266, in _create_dicts_from_changes
>     _store_values(block_and_field_name, value, values)
>   File "/nsls2/users/dgavrilov/repos/PandABlocks-ioc/src/pandablocks_ioc/ioc.py", line 228, in _store_values
>     number_of_blocks = block_info[block_name_number[:-1]].number
> KeyError: 'TTLOUT1'
> ```

Thanks for catching this, this bug should be fixed now?

**Comment by dmgav** [comment]
I think it would make sense to fix it here. It should be a small change. I can try debugging it later today with Panda.

**Comment by evalott100** [comment]
> I think it would make sense to fix it here. It should be a small change. I can try debugging it later today with Panda.

I've reproduced the error should be fixed from [72236b0](https://github.com/PandABlocks/PandABlocks-ioc/pull/30/commits/72236b0d88d54845587ee6c19c8635686f69b24a)

File: src/pandablocks_ioc/ioc.py [hunk 3]
```diff
@@ -222,18 +223,19 @@ def _store_values(
...
-            # The block is fixed with metadata
+            # The block is fixed with metadata, it should end with a number
             #     "*METADATA.LABEL_SEQ2": "NewSeqMetadataLabel",
             if not block_name_number[-1].isdigit():
                 raise ValueError(
                     f"Recieved metadata for a block name {block_name_number} that "
                     "didn't contain a number"
                 )
+
+            parts = re.findall(r"\d+|[^\d]+", block_name_number)
```
Comment (dmgav) [inline src/pandablocks_ioc/ioc.py:235]: I also debugged the issue. The original code worked well for blocks with less than 10 instances. The following code would expand it to work with blocks with 2 digit numbers. The block names may contain numbers in the middle, for example `SFP2_SYNC_OUT`.
```
block_name_no_number = block_name_number if block_name_number in block_info else re.sub(r"\d*$", "", block_name_number)
```

Comment (evalott100) [reply src/pandablocks_ioc/ioc.py:235]: Yep good point! I'll make the change in dev before merging to main.

**Pull request merged**